### PR TITLE
Added implementations for get_token_largest_accounts() and get_token_supply()

### PR DESCRIPTION
**Pull request by whdev1**
I have added implementations for the get_token_largest_accounts() and get_token_supply() methods in the synchronous RPC client as well as their respective _args methods in rpc.core.

### CODE REVIEW

**Comment by michaelhly** [comment]
Would you mind writing a couple integration test for these?

**Comment by whdev1** [comment]
I've gone ahead and fixed the formatting issues and this latest commit is passing with Black. I also wrote integration tests for both methods.

I created a stubbed_token_addr() method for testing as both of the new functions require a known valid token address to return results instead of an error.

File: tests/conftest.py [hunk 1]
```diff
@@ -111,6 +111,12 @@ def async_stubbed_sender_cached_blockhash() -> Account:
     return Account(bytes([3] * PublicKey.LENGTH))
 
 
+@pytest.fixture(scope="session")
+def stubbed_token_addr() -> PublicKey:
+    """An arbitrary known token address."""
+    return PublicKey("DeV3CnJsrhN5TzCZQvmuQgYn1Up8nDycqmzxZJWH9DAw")
+
+
```
Comment (michaelhly) [inline tests/conftest.py:119]:
```suggestion
```

File: tests/integration/test_http_client.py [hunk 2]
```diff
@@ -490,6 +490,20 @@ def test_get_multiple_accounts(stubbed_sender, test_http_client):
     assert_valid_response(resp)
 
 
+@pytest.mark.integration
+def test_get_token_largest_accounts(stubbed_token_addr, test_http_client):
+    """Test get token largest accounts."""
+    resp = test_http_client.get_token_largest_accounts(stubbed_token_addr.public_key())
```
Comment (michaelhly) [inline tests/integration/test_http_client.py:496]:
```suggestion
    resp = test_http_client.get_token_largest_accounts(WRAPPED_SOL_MINT)
```

File: tests/integration/test_http_client.py [hunk 3]
```diff
@@ -490,6 +490,20 @@ def test_get_multiple_accounts(stubbed_sender, test_http_client):
...
+    assert_valid_response(resp)
+
+
+@pytest.mark.integration
+def test_get_token_supply(stubbed_token_addr, test_http_client):
+    """Test get token largest accounts."""
+    resp = test_http_client.get_token_supply(stubbed_token_addr.public_key())
```
Comment (michaelhly) [inline tests/integration/test_http_client.py:503]:
```suggestion
    resp = test_http_client.get_token_supply(WRAPPED_SOL_MINT)
```

**Review by michaelhly — COMMENTED** [review]
Please use this mint instead:
https://michaelhly.github.io/solana-py/spl.html#spl.token.constants.WRAPPED_SOL_MINT

**Comment by whdev1** [comment]
Thank you, was not aware of the constant. I went ahead and made all of the suggested changes in this new commit and added an import for WRAPPED_SOL_MINT in test_http_client.py

**Comment by kevinheavey** [comment]
@whdev1 Would you mind adding the async methods and tests too? If you're not familiar with async stuff in Python, you can look at async_api.py and test_async_http_client.py and just mimic those. Will just be a matter of copy-pasting and changing a few characters (Python forces a bit of duplication when it comes to sync/async stuff)

**Comment by whdev1** [comment]
@kevinheavey Definitely. I just added a new commit with the async implementations for both. I've also changed the title of the PR to reflect this.

File: tests/integration/test_async_http_client.py [hunk 4]
```diff
@@ -500,6 +501,20 @@ async def test_get_supply(test_http_client_async):
     assert_valid_response(resp)
 
 
+@pytest.mark.integration
```
Comment (kevinheavey) [inline tests/integration/test_async_http_client.py:504]: Need too `@pytest.mark.asyncio` here

File: tests/integration/test_async_http_client.py [hunk 5]
```diff
@@ -500,6 +501,20 @@ async def test_get_supply(test_http_client_async):
...
+async def test_get_token_largest_accounts(test_http_client):
+    """Test get token largest accounts."""
+    resp = test_http_client.get_token_largest_accounts(WRAPPED_SOL_MINT)
```
Comment (kevinheavey) [inline tests/integration/test_async_http_client.py:507]: Should be `resp = await test_http_client_async.get_token_largest_accounts(WRAPPED_SOL_MINT)`

**Comment by whdev1** [comment]
Thank you, just fixed both of those issues.

File: tests/integration/test_async_http_client.py [hunk 6]
```diff
@@ -500,6 +501,22 @@ async def test_get_supply(test_http_client_async):
     assert_valid_response(resp)
 
 
+@pytest.mark.integration
+@pytest.mark.asyncio
+async def test_get_token_largest_accounts(test_http_client_async):
+    """Test get token largest accounts."""
+    resp = test_http_client_async.get_token_largest_accounts(WRAPPED_SOL_MINT)
```
Comment (kevinheavey) [inline tests/integration/test_async_http_client.py:508]:
```suggestion
    resp = await test_http_client_async.get_token_largest_accounts(WRAPPED_SOL_MINT)
```

File: tests/integration/test_async_http_client.py [hunk 7]
```diff
@@ -500,6 +501,22 @@ async def test_get_supply(test_http_client_async):
...
+    assert_valid_response(resp)
+
+
+@pytest.mark.integration
+@pytest.mark.asyncio
+async def test_get_token_supply(test_http_client_async):
+    """Test get token supply."""
+    resp = test_http_client_async.get_token_supply(WRAPPED_SOL_MINT)
```
Comment (kevinheavey) [inline tests/integration/test_async_http_client.py:516]:
```suggestion
    resp = await test_http_client_async.get_token_supply(WRAPPED_SOL_MINT)
```

**Comment by kevinheavey** [comment]
lgtm, @michaelhly does it look good to you?

**Review by michaelhly — APPROVED** [review]
LGTM.

**Pull request merged**